website/docs/developer-guide/chaoscenter-developer-guide.md

```diff
@@ -0,0 +1,219 @@
+---
+id: chaoscenter-developer-guide
+title: ChaosCenter Developer Guide
+sidebar_label: ChaosCenter Developer Guide 
+---
+
+---
+
+## **Prerequisites**
+:::note
+This document is ietented to be implemented locally. Please do not use in dev or prod environmentss
+:::
+
+- Kubernetes 1.17 or later
+- Helm3 or Kubectl
+- Node and npm
+- Docker
+- Golang
+- Local Kubernetes Cluster (via minikube, k3s or kind)
+
+## **Control Plane**
+Backend components consist of three microservices
+1. GraphQL-Server
+2. Authentication-Server
+3. MongoDB
+
+Frontend component
+1. React
+
+## **Steps to run the Control Plane**
+
+### 1. Run MongoDB
+
+Step-1: Pull and run the image 
+
+```bash
+docker pull mongo:5
+
+docker network create mongo-cluster 
+
+docker run -d --net mongo-cluster -p 27015:27015 --name m1 mongo:4.2 mongod --replSet rs0 --port 27015 
+docker run -d --net mongo-cluster -p 27016:27016 --name m2 mongo:4.2 mongod --replSet rs0 --port 27016
+docker run -d --net mongo-cluster -p 27017:27017 --name m3 mongo:4.2 mongod --replSet rs0 --port 27017
+```
+
+Step-2: Add hosts 
+
+Windows
+```bash
+# add hosts in hosts 
+notepad C:\Windows\System32\drivers\etc\hosts
+
+# add the below line
+127.0.0.1       m1 m2 m3
+```
+
+Linux/Mac
+```bash
+# add hosts in hosts 
+sudo vim /etc/hosts
+
+# add the below line
+127.0.0.1       m1 m2 m3
+```
+
+
+Step-3: Configure the mongoDB replica set
+
+```bash
+docker exec -it m1 mongo -port 27015
+
+config={"_id":"rs0","members":[{"_id":0,"host":"m1:27015"},{"_id":1,"host":"m2:27016"},{"_id":2,"host":"m3:27017"}]}
+
+rs.initiate(config)
+
+db.getSiblingDB("admin").createUser({user:"admin",pwd:"1234",roles:[{role:"root",db:"admin"}]});
+```
```

Comment:
We should use Docusaurus tabs for individual infra types to show them. Please refactor in all the places.